qemu/tests/ovs_host_vlan.py

```diff
@@ -0,0 +1,298 @@
+import logging
+import os.path
+
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import remote
+from virttest import data_dir
+from virttest import utils_net
+from virttest import utils_test
+from virttest import env_process
+
+
+def create_file_in_guest(test, session, file_path, size=100, os_type="linux", timeout=360):
+    """
+    Create a file with 'dd' in guest.
+
+    :param test: QEMU test object
+    :param session: in which guest to do the file creation
+    :param file_path: put the created file in which directory
+    :param size: file size
+    :param os_type: linux or windows
+    :param timeout: file creation command timeout
+    """
+    size = int(size)
+    if os_type == "linux":
+        cmd = "dd if=/dev/zero of=%s bs=1M count=%d" % (file_path, size)
+    else:
+        cmd = "fsutil file createnew %s %s" % (file_path, size * 1024 * 1024)
+    status, output = session.cmd_status_output(cmd, timeout=timeout)
+    if status:
+        err = "Fail to create file in guest."
+        err += " command '%s', output '%s'." % (cmd, output)
+        test.error(err)
+
+
+def ping_result_check(test, loss_ratio, same_vlan):
+    """
+    Ping result check.
+
+    :param test: QEMU test object
+    :param loss_ratio: ping test loss ratio
+    :param same_vlan: whether the two guests are in the same vlan
+    """
+    if same_vlan and loss_ratio > 0:
+        msg = "Package lost when ping guest in same vlan."
+        msg += "Loss ratio is %s" % loss_ratio
+        test.fail(msg)
+    if not same_vlan and loss_ratio != 100:
+        msg = "Ping between guest in different vlan successful."
+        msg += "Loss ratio is %s" % loss_ratio
+        test.fail(msg)
+
+
+def ping(test, os_type, match_error, dest, count, session, same_vlan):
+    """
+    In 'session' ping 'dest'.
+    If the two guests are in the same vlan, loss ratio should be 0%.
+    Otherwise, loss ratio should be 100%.
+
+    :param test: QEMU test object
+    :param dest: dest ip address
+    :param count: ping count
+    :param session: in which guest to do ping test
+    :param same_vlan: whether the two guests are in the same vlan
+    """
+    if os_type == "linux":
+        #ping_cmd = "ping %s -c %s" % (dest, count)
+        #status, output = session.cmd_status_output(ping_cmd, timeout=int(count) * 1.5)
+        status, output = utils_test.ping(dest, count, timeout=int(count) * 1.50,
+                                         session=session)
+        loss_ratio = utils_test.get_loss_ratio(output)
+        ping_result_check(test, loss_ratio, same_vlan)
+        logging.debug("%s" % output)
+    elif os_type == "windows":  # TBD, not supported by now #
+        ping_cmd = "ping %s -n %s" % (dest, count)
+        status, output = session.cmd_status_output(ping_cmd, timeout=60)
```

Comment:
timeout is different.

Reply:
@xiywang That could be controlled by parameters, instead of hard code the value.

Reply:
@xiywang This line actually can be determined for now, the only uncertain part is timeout value, which should be configurable, so it can be listed in Windows params, so the ping() action could be removed from the if condition(TODO part). And that will also slightly reduce the adjustment when you support Windows later.
Anyway I have no strong opinion against this, it's all up to you.